test/functional/bitcoin_cli.py

```diff
@@ -35,5 +35,30 @@ def run_test(self):
         assert_equal(["foo", "bar"], self.nodes[0].cli('-rpcuser=%s' % user, '-stdin', '-stdinrpcpass', input=password + "\nfoo\nbar").echo())
         assert_raises_process_error(1, "incorrect rpcuser or rpcpassword", self.nodes[0].cli('-rpcuser=%s' % user, '-stdin', '-stdinrpcpass', input="foo").echo)
 
+        self.log.info("Compare responses from `bitcoin-cli -getinfo` and the RPCs data is retrieved from.")
+        cli_get_info = self.nodes[0].cli('-getinfo').help()
+        wallet_info = self.nodes[0].getwalletinfo()
+        network_info = self.nodes[0].getnetworkinfo()
+        blockchain_info = self.nodes[0].getblockchaininfo()
+
+        assert_equal(cli_get_info['version'], network_info['version'])
+        assert_equal(cli_get_info['protocolversion'], network_info['protocolversion'])
+        assert_equal(cli_get_info['walletversion'], wallet_info['walletversion'])
+        assert_equal(cli_get_info['balance'], wallet_info['balance'])
+        assert_equal(cli_get_info['blocks'], blockchain_info['blocks'])
+        assert_equal(cli_get_info['timeoffset'], network_info['timeoffset'])
+        assert_equal(cli_get_info['connections'], network_info['connections'])
+        assert_equal(cli_get_info['proxy'], network_info['networks'][0]['proxy'])
+        assert_equal(cli_get_info['difficulty'], blockchain_info['difficulty'])
+        assert_equal(cli_get_info['testnet'], blockchain_info['chain'] == "test")
+        assert_equal(cli_get_info['balance'], wallet_info['balance'])
+        assert_equal(cli_get_info['keypoololdest'], wallet_info['keypoololdest'])
+        assert_equal(cli_get_info['keypoolsize'], wallet_info['keypoolsize'])
+        assert_equal(cli_get_info['paytxfee'], wallet_info['paytxfee'])
+        assert_equal(cli_get_info['relayfee'], network_info['relayfee'])
+
+        if 'unlocked_until' in wallet_info:
```

Comment:
We know that the wallet isn't locked, so this won't be tested. I don't think there's any benefit in adding a test code that won't be executed (in fact it's slightly deleterious since a casual reader might think that we're actually testing this).

Either:
- lock the wallet and actually test this
- add a comment saying we're not testing the field because the wallet isn't locked.

I think either is fine.

Reply:
Done